### PR TITLE
Initial Profobuf messages

### PR DESCRIPTION
**Pull request by doomspork**
A starting point for our conversation on leveraging Protobuf messages in our events.

### CODE REVIEW

**Comment by doomspork** [comment]
In the interest of not going hogwild with CI/CD — yet — we're going to need to manually generate the Elixir files when updating the Protobuf files.

**Comment by doomspork** [comment]
We should also discuss whether or not we want to wrap all of our messages in an "Envelope" that contains meta data we might want (timestamps, source, et al).

**Comment by btkostner** [comment]
mhmmm, we should at least create a mix task / alias to run that command. Are we able to override the built in `mix compile` command to also run that command or would that be horrible / impossible?

**Comment by btkostner** [comment]
Also, if we are generating elixir in this repo, why not just have elixir code without the protobuf stuff? If we are planning to use this across languages, we should make this repo _just_ the protobuf stuff and do all of the elixir stuff in what ever project uses this.

File: mix.exs [hunk 1]
```diff
@@ -24,4 +25,11 @@ defmodule Bottle.MixProject do
       {:protobuf, "~> 0.7.1"}
     ]
   end
+
+  defp aliases do
+    [
+      "protobuf.gen": ["protoc --elixir_out=./lib/bottle -I priv/protos/ priv/protos/*.proto
```
Comment (btkostner) [inline mix.exs:31]: :parrotwave1: :parrotwave2: :parrotwave3: :parrotwave4: :parrotwave5: :parrotwave6: :parrotwave7: :parrotwave8: :parrotwave9:

File: .formatter.exs [hunk 2]
```diff
@@ -0,0 +1,4 @@
+# Used by "mix format"
+[
+  inputs: ["{mix,.formatter}.exs", "{config,lib,test}/**/*.{ex,exs}"]
```
Comment (btkostner) [inline .formatter.exs:3]: We can drop this and add it to the elixir branch if need be

File: README.md [hunk 3]
```diff
@@ -1,2 +1,19 @@
...
+```shell
+$ protoc --elixir_out=./lib/bottle/messages -I priv/protos/ priv/protos/*.proto
+```
+
+## Installation
+
+```elixir
+def deps do
+  [
+    {:bottle, github: "system76/bottle"}
```
Comment (btkostner) [inline README.md:16]:
```suggestion
    {:bottle, github: "system76/bottle", branch: "elixir"}
```

File: .formatter.exs (hunk 2 above)
Comment (doomspork) [reply .formatter.exs]: Just did :+1:

**Pull request merged**